### DISCUSSION TITLE
Polling k8s objects for change

### DISCUSSION THREAD

**Discussion by allensiho** [dotnet/dotnet-operator-sdk#773] [Q&A]
Hi,

I wanted to annotate some pods and say add a corresponding hpa for that pod

Is there a way to have watchers in your controller that can detect when a pod is annotated or have an notation removed to create/remove has?

**Comment by buehler** [comment]
Hey @allensiho 

I don't quite remember if changes to annotations do trigger a reconciliation. If it is the case, then yes, a normal controller should be aware of it.

If not, you can always implement your own logic with a background service that periodically fetches some objects and does stuff with them.

**Reply by allensiho** [reply]
Oh, so not just change to your CRD object can trigger reconciliation?

You don't have to subscribe to them?

There are other events to?

Not sure if there is a clear tutorial somewhere 

Thanks
